romancal/skycell/match.py

```diff
@@ -26,88 +26,177 @@ class ImageFootprint:
 
-    def __init__(self, radec_corners: list[tuple[float, float]]):
+    def __init__(self, radec_vertices: list[tuple[float, float]]):
         """
         Parameters
         ----------
-        radec_corners: list[tuple[float, float]]
-            four corners of the image in right ascension and declination
+        radec_vertices: list[tuple[float, float]]
+            vertices (usually the corners) of the image in right ascension and declination
         """
-        radec_corners = np.array(radec_corners)
-        if radec_corners.shape[0] != 4:
-            raise ValueError(f"need 4 corners, not {radec_corners.shape[0]}")
-        self._radec_corners = radec_corners
+        self._radec_vertices = np.array(radec_vertices)
 
     @classmethod
-    def from_gwcs(
-        cls, iwcs: WCS, image_shape: tuple[int, int] | None = None
+    def from_wcs(
+        cls,
+        wcs: WCS,
+        vertices: tuple[int, int] = (0, 0),
     ) -> "ImageFootprint":
-        """create an image footprint from a GWCS object (and image shape, if no bounding box is present)"""
-
-        if not hasattr(iwcs, "bounding_box") or iwcs.bounding_box is None:
-            if image_shape is None:
-                # pixel_shape is in x, y order contrary to numpy convention.
-                if hasattr(iwcs, "pixel_shape") and iwcs.pixel_shape is not None:
-                    image_shape = (iwcs.pixel_shape[1], iwcs.pixel_shape[0])
-                else:
-                    raise ValueError(
-                        "Cannot infer image footprint from GWCS object because "
-                        "`image_shape` was not specified and "
-                        "the GWCS object does not have `.bounding_box` nor `.pixel_shape`."
-                    )
+        """
+        create an image footprint from a GWCS object (and image shape, if no bounding box is present)
+
+        Parameters
+        ----------
+
+        wcs: WCS object
+        vertices: number of vertices to create on each edge in addition to the corner pixels
+        """
+
+        # if `.pixel_shape` is not defined, try deriving it from the `.bounding_box`
+        if not hasattr(wcs, "pixel_shape") or wcs.pixel_shape is None:
+            if hasattr(wcs, "bounding_box") and wcs.bounding_box is not None:
+                wcs.pixel_shape = wcs_pixel_shape_from_bbox(wcs.bounding_box)
+            else:
+                raise ValueError(
+                    "Cannot infer image footprint from WCS without `.pixel_shape` nor `.bounding_box`."
+                )
 
-            iwcs.bounding_box = wcs_bbox_from_shape(image_shape)
+        # pixel_shape is in x, y order contrary to numpy convention.
+        image_shape = (wcs.pixel_shape[1], wcs.pixel_shape[0])
+
+        if vertices != (0, 0):
+            if not hasattr(wcs, "bounding_box") or wcs.bounding_box is None:
+                wcs.bounding_box = wcs_bbox_from_shape(image_shape)
+
+            vertex_points = wcs.footprint(center=False)
```

Comment:
Does this say: if the user specified the vertices, ignore them and replace them with wcs.footprint(...)?  Maybe the intent is "if vertices == (0, 0)"?

Reply:
ah, good catch!